server/checks/Approval.js

```diff
@@ -13,17 +14,125 @@ export default class Approval extends Check {
   static NAME = 'Approval check'
   static HOOK_EVENTS = [EVENTS.PULL_REQUEST, EVENTS.ISSUE_COMMENT]
 
-  static generateStatusMessage(actual, needed) {
-    if (actual < needed) {
-      return `This PR needs ${needed - actual} more approvals (${actual}/${needed} given).`
+  static generateStatus(approvals, {minimum, groups}) {
+    if (Object.keys(approvals.groups || {}).length > 0) {
+      // check group requirements
+      const unsatisfied = Object.keys(approvals.groups)
+                                .map(approvalGroup => {
+                                  const needed = groups[approvalGroup].minimum
+                                  const given = approvals.groups[approvalGroup]
+                                  const diff = needed - given
+                                  if (diff > 0) {
+                                    return {approvalGroup, diff, needed, given}
+                                  } else {
+                                    return false
+                                  }
+                                })
+                                .filter(diff => !!diff)
```

Comment:
Instead of map+filter we can just use reduce with an array to which we append or not.